src/login/AzureLoginHelper.ts

```diff
@@ -73,21 +74,27 @@ export class AzureAccountLoginHelper {
 			if (e.affectsConfiguration(getSettingWithPrefix(resourceFilterSetting))) {
 				actionContext.telemetry.properties.changeResourceFilter = 'true';
 				updateFilters(true);
-			} else if (e.affectsConfiguration(getSettingWithPrefix(authLibrarySetting))) {
-				actionContext.telemetry.properties.changeAuthLibrary = 'true';
-	
-				const newAuthLibrary: AuthLibrary = getAuthLibrary();
-				await this.context.globalState.update(authLibraryCacheKey, { lastUsedAuthLibrary: newAuthLibrary });
-
-				const mustSignOutAndReload: string = localize('azure-account.mustSignOutAndReload', 'You must sign out and reload the window to authenticate with "{0}"', newAuthLibrary);
-				const signOutAndReload: string = localize('azure-account.signOutAndReload', 'Sign Out and Reload Window');
-				void window.showInformationMessage(mustSignOutAndReload, signOutAndReload).then(async value => {
-					if (value === signOutAndReload) {
-						actionContext.telemetry.properties.signOutAndReload = 'true';
-						await this.logout(true /* forceLogout */);
-						await commands.executeCommand('workbench.action.reloadWindow');
+			} else if (e.affectsConfiguration(getSettingWithPrefix(authLibrarySetting)) || e.affectsConfiguration(getSettingWithPrefix(cloudSetting)) || e.affectsConfiguration(getSettingWithPrefix(tenantSetting))) {
```

Comment:
I'd alter this to check for anything within `cachedSettingKeys`. Even though we check each of the three individually on lines 87 and 93, since the purpose of this outer guard is to catch everything in that `cachedSettingKeys` array, I think it's best to do that.

e.g.
```suggestion
			} else if (cachedSettingKeys.some(k => e.affectsConfiguration(getSettingWithPrefix(k)))) {
```